cfme/infrastructure/pxe.py

```diff
@@ -442,51 +446,18 @@ class PXECustomizationTemplateEditView(PXECustomizationTemplateForm):
     cancel = Button('Cancel')
 
 
-class CustomizationTemplate(Updateable, Pretty, Navigatable):
-    """ Model of a Customization Template in CFME
-
-    Args:
-        name: The name of the template.
-        description: Template description.
-        image_type: Image type name, must be one of an existing System Image Type.
-        script_type: Script type, either Kickstart, Cloudinit or Sysprep.
-        script_data: The scripts data.
+@attr.s
+class CustomizationTemplate(Updateable, Pretty, BaseEntity):
+    """
+    Model of a Customization Template in CFME
     """
     pretty_attrs = ['name', 'image_type']
 
-    def __init__(self, name=None, description=None, image_type=None, script_type=None,
-                 script_data=None, appliance=None):
-        Navigatable.__init__(self, appliance=appliance)
-        self.name = name
-        self.description = description
-        self.image_type = image_type
-        self.script_type = script_type
-        self.script_data = script_data
-
-    def create(self, cancel=False):
-        """
-        Creates a Customization Template object
-
-        Args:
-            cancel (boolean): Whether to cancel out of the creation.  The cancel is done
-                after all the information present in the CT has been filled in the UI.
-        """
-        view = navigate_to(self, 'Add')
-
-        view.fill({'name': self.name,
-                   'description': self.description,
-                   'image_type': self.image_type,
-                   'type': self.script_type,
-                   'script': self.script_data})
-        main_view = self.create_view(PXECustomizationTemplatesView)
-
-        if cancel:
-            view.cancel.click()
-            msg = 'Add of new Customization Template was cancelled by the user'
-        else:
-            view.add.click()
-            msg = 'Customization Template "{}" was saved'.format(self.name)
-        main_view.flash.assert_success_message(msg)
+    name = attr.ib()
```

Comment:
**Question/Required**

`name`, `description`, and `script_type` were optional args before, why are they now required? Have all uses been updated?